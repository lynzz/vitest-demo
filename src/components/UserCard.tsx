import React from 'react';

interface User {
  id: number;
  name: string;
  email: string;
  avatar?: string;
}

interface UserCardProps {
  user: User;
  isActive?: boolean;
}

export const UserCard: React.FC<UserCardProps> = ({ 
  user, 
  isActive = false 
}) => {
  return (
    <div className={`card ${isActive ? 'active' : ''}`}>
      <div className="card-header">
        {user.avatar && (
          <img src={user.avatar} alt={user.name} className="avatar" />
        )}
        <h3>{user.name}</h3>
      </div>
      <div className="card-body">
        <p className="email">{user.email}</p>
        <div className="status">
          {isActive ? 'ðŸŸ¢ åœ¨çº¿' : 'âš« ç¦»çº¿'}
        </div>
      </div>
    </div>
  );
}; 