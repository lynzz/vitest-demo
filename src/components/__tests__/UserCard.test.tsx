import { describe, test, expect } from 'vitest'
import { render } from '@testing-library/react'
import { UserCard } from '../UserCard'

const mockUser = {
  id: 1,
  name: 'John Doe',
  email: 'john@example.com',
  avatar: 'https://example.com/avatar.jpg'
};

describe('UserCard Snapshots', () => {
  test('renders default state correctly', () => {
    const { container } = render(<UserCard user={mockUser} />);
    
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders active state correctly', () => {
    const { container } = render(
      <UserCard user={mockUser} isActive={true} />
    );
    
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders without avatar correctly', () => {
    const userWithoutAvatar = { ...mockUser, avatar: undefined };
    const { container } = render(<UserCard user={userWithoutAvatar} />);
    
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches inline snapshot', () => {
    const { container } = render(<UserCard user={mockUser} />);
    expect(container.firstChild).toMatchInlineSnapshot(`
      <div
        class="card "
      >
        <div
          class="card-header"
        >
          <img
            alt="John Doe"
            class="avatar"
            src="https://example.com/avatar.jpg"
          />
          <h3>
            John Doe
          </h3>
        </div>
        <div
          class="card-body"
        >
          <p
            class="email"
          >
            john@example.com
          </p>
          <div
            class="status"
          >
            ⚫ 离线
          </div>
        </div>
      </div>
    `);
  });

  // 传统的属性测试（非快照）
  test('displays user name correctly', () => {
    const { getByText } = render(<UserCard user={mockUser} />);
    
    expect(getByText('John Doe')).toBeInTheDocument();
  });

  test('displays user email correctly', () => {
    const { getByText } = render(<UserCard user={mockUser} />);
    
    expect(getByText('john@example.com')).toBeInTheDocument();
  });

  test('shows online status when active', () => {
    const { getByText } = render(<UserCard user={mockUser} isActive={true} />);
    
    expect(getByText('🟢 在线')).toBeInTheDocument();
  });

  test('shows offline status when inactive', () => {
    const { getByText } = render(<UserCard user={mockUser} isActive={false} />);
    
    expect(getByText('⚫ 离线')).toBeInTheDocument();
  });

  test('applies active class when isActive is true', () => {
    const { container } = render(<UserCard user={mockUser} isActive={true} />);
    
    expect(container.firstChild).toHaveClass('card', 'active');
  });

  test('does not apply active class when isActive is false', () => {
    const { container } = render(<UserCard user={mockUser} isActive={false} />);
    
    expect(container.firstChild).toHaveClass('card');
    expect(container.firstChild).not.toHaveClass('active');
  });
}); 