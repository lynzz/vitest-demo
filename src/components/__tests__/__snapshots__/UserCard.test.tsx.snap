// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserCard Snapshots > renders active state correctly 1`] = `
<div
  class="card active"
>
  <div
    class="card-header"
  >
    <img
      alt="John Doe"
      class="avatar"
      src="https://example.com/avatar.jpg"
    />
    <h3>
      John Doe
    </h3>
  </div>
  <div
    class="card-body"
  >
    <p
      class="email"
    >
      john@example.com
    </p>
    <div
      class="status"
    >
      ðŸŸ¢ åœ¨çº¿
    </div>
  </div>
</div>
`;

exports[`UserCard Snapshots > renders default state correctly 1`] = `
<div
  class="card "
>
  <div
    class="card-header"
  >
    <img
      alt="John Doe"
      class="avatar"
      src="https://example.com/avatar.jpg"
    />
    <h3>
      John Doe
    </h3>
  </div>
  <div
    class="card-body"
  >
    <p
      class="email"
    >
      john@example.com
    </p>
    <div
      class="status"
    >
      âš« ç¦»çº¿
    </div>
  </div>
</div>
`;

exports[`UserCard Snapshots > renders without avatar correctly 1`] = `
<div
  class="card "
>
  <div
    class="card-header"
  >
    <h3>
      John Doe
    </h3>
  </div>
  <div
    class="card-body"
  >
    <p
      class="email"
    >
      john@example.com
    </p>
    <div
      class="status"
    >
      âš« ç¦»çº¿
    </div>
  </div>
</div>
`;
